packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -0,0 +1,468 @@
+import * as sc from "./FolderDataGrid.sc";
+import { Footer } from "./footer/Footer";
+import { DamItemLabelColumn } from "./label/DamItemLabelColumn";
+import { MoveDamItemDialog } from "./moveDialog/MoveDamItemDialog";
+import { moveDamFilesMutation, moveDamFoldersMutation } from "./moveDialog/MoveDamItemDialog.gql";
+import { TableHead } from "./TableHead";
+import { useDamSearchHighlighting } from "./useDamSearchHighlighting";
+
+export type FooterType = "upload" | "selection";
+interface FooterInfo {
+    type: FooterType;
+
+    folderName?: string;
+}
+
+export type DamItemSelectionMap = Map<string, "file" | "folder">;
+
+interface FolderDataGridProps extends DamConfig {
+    id?: string;
+    breadcrumbs?: BreadcrumbItem[];
+    filterApi: IFilterApi<DamFilter>;
+    selectionApi: ISelectionApi;
+}
+
+export const FolderDataGrid = ({
+    id,
+    filterApi,
+    breadcrumbs,
+    selectionApi,
+    hideContextMenu,
+    hideArchiveFilter,
+    hideMultiselect,
+    renderDamLabel,
+    TableHeadActionButton,
+    ...props
+}: FolderDataGridProps): React.ReactElement => {
+    const intl = useIntl();
+    const apolloClient = useApolloClient();
+
+    const [selectionMap, setSelectionMap] = React.useState<DamItemSelectionMap>(new Map());
+    const [footerInfo, setFooterInfo] = React.useState<FooterInfo | null>(null);
+
+    const showFooter = useThrottledCallback((type: FooterType, specificInfo?: { folderName?: string }) => {
+        setFooterInfo({
+            type,
+            folderName: specificInfo?.folderName,
+        });
+    }, 500);
+
+    const hideFooter = React.useCallback(() => {
+        if (showFooter.isPending()) {
+            showFooter.cancel();
+        }
+
+        if (footerInfo?.type === "upload" && selectionMap.size > 0) {
+            showFooter("selection");
+            return;
+        }
+
+        setFooterInfo(null);
+    }, [footerInfo?.type, selectionMap.size, showFooter]);
+
+    React.useEffect(() => {
+        if (selectionMap.size > 0) {
+            showFooter("selection");
+        } else {
+            hideFooter();
+        }
+
+        // useEffect should only be executed if the selection changes
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectionMap]);
+
+    const [pageSize, setPageSize] = useStoredState<number>("FolderDataGrid-pageSize", 20);
+    const pagingApi = useTableQueryPaging(0, { persistedStateId: "FolderDataGrid-pagingApi" });
+
+    const { data } = useQuery<GQLDamFolderQuery, GQLDamFolderQueryVariables>(damFolderQuery, {
+        variables: {
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            id: id!,
+        },
+        skip: id === undefined,
+    });
+
+    const {
+        tableData,
+        loading: tableLoading,
+        error,
+    } = useTableQuery<GQLDamItemsListQuery, GQLDamItemsListQueryVariables>()(damItemsListQuery, {
```

Comment:
When switching to DataGrid I would not reccomend using the Table hooks from comet (useTableQuery, useTableQueryPaging, useTableQueryFilter) as they are optimized for comet Table. One day Table and useTableQuery will get deprecated.

And we also have a few helper hooks in comet for DataGrid (useDataGridRemote, usePersistentColumnState) that you can consider using (if you haven't already)

See also this example: https://github.com/vivid-planet/comet/blob/main/demo/admin/src/products/ProductsTable.tsx